src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Performance/RecommendCaseInsensitiveStringComparison.Base.Tests.cs

```diff
@@ -397,6 +514,32 @@ public static IEnumerable<object[]> NoDiagnosticData()
             yield return new object[] { "\"aBc\".CompareTo(\"cDe\")" };
         }
 
+        public static IEnumerable<object[]> DiagnosticNoFixStartsWithContainsIndexOfData()
```

Comment:
Just to confirm - the three new methods added here and after the primary ones validating diagnostic but no code fix case, while most of the updates above in the test file are adding to the existing test matrix where you are validating ordering fixup of named arguments to match the original parameter order.

Thank for the extensive tests and detailed comments. Some of these examples might be helpful within the code fixer code as well.

Reply:
You almost got it! Yes, I added new tests that confirm that we offer a diagnostic but no fix. For that, I had to use the same TestCode and FixedCode in the helper method that verifies the before and after.

I also edited a few test cases so that they stop offering a fix (the cases with ToUpper vs ToUpperInvariant).

And I also added many more tests that I realized I was missing.